src/ctx.rs

```diff
@@ -731,6 +731,29 @@ impl<'a> TryIntoCtx for &'a [u8] {
     }
 }
 
+impl<'a> TryFromCtx<'a, usize> for &'a [u8] {
+    type Error = error::Error;
+    #[inline]
+    fn try_from_ctx(src: &'a [u8], size: usize) -> result::Result<(Self, usize), Self::Error> {
+        if size > src.len() {
+            Err(error::Error::TooBig {
+                size,
+                len: src.len(),
+            })
+        } else {
+            Ok((&src[..size], size))
+        }
+    }
+}
+
+impl<'a> TryFromCtx<'a> for &'a [u8] {
```

Comment:
so the problem with having two contexts is that now e.g., this function call becomes ambiguous:
```rust
    const buf: &[u8] = &[0x1, 0x2, 0x3, 0x4];
    let res = buf.pread::<&[u8]>(0);
```
You should see an error like this after a bunch of generic vomit:
```
error[E0283]: type annotations needed
   --> tests/api.rs:388:19
    |
388 |     let res = buf.pread::<&[u8]>(0);
    |                   ^^^^^
    |
    = note: multiple `impl`s satisfying `&[u8]: TryFromCtx<'_, _>` found in the `scroll` crate:
            - impl<'a> TryFromCtx<'a, usize> for &'a [u8];
            - impl<'a> TryFromCtx<'a> for &'a [u8];
```

I'm very surprised we do not have a test in either `api.rs` or somewhere in `lib.rs` to ensure that existing `preads` of e.g., `&[u8]` are compilable. I believe introducing this could theoretically break downstream users as any code that did a `pread` of `&[u8]` will no longer compile.

I swear we had tests specifically to test that just `pread` worked fine on `&[u8]` but don't see any tests failing...